test/functional/pil.js

```diff
@@ -0,0 +1,178 @@
+      return results;
+    };
+
+    const isNotBillable = (results, email) => {
+      const emails = results.map(pil => pil.profile.email);
+      assert.ok(!emails.includes(email), `Billable PIL list should not contain PIL for ${email}`);
+      return results;
+    };
+
+    it('exposes a `billable` query method', () => {
+      const { PIL } = this.models;
+      return assert.equal(typeof PIL.query().billable, 'function');
+    });
+
+    it('includes PILs which were valid in the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 102, start: '2018-04-06', end: '2019-04-05' })
+        .then(results => isBillable(results, 'activepil@example.com'));
+    });
+
+    it('does not include PILs which were revoked before the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 102, start: '2018-04-06', end: '2019-04-05' })
+        .then(results => isNotBillable(results, 'revokedpil@example.com'));
+    });
+
+    it('does not include PILs which were granted after the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 102, start: '2016-04-06', end: '2017-04-05' })
+        .then(results => isNotBillable(results, 'activepil@example.com'));
+    });
+
+    it('does not include PILs which were transfered into the establishment after the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 102, start: '2018-04-06', end: '2019-04-05' })
+        .then(results => isNotBillable(results, 'transferedpil@example.com'));
+    });
+
+    it('includes PILs which were transfered out of the establishment after the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 101, start: '2018-04-06', end: '2019-04-05' })
+        .then(results => isBillable(results, 'transferedpil@example.com'));
+    });
+
+    it('includes PILs which were transferred out of the establishment during the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 101, start: '2019-04-06', end: '2020-04-05' })
+        .then(results => isBillable(results, 'transferedpil@example.com'));
+    });
+
+    it('includes PILs which were transferred in and out of the establishment during the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 100, start: '2018-04-06', end: '2019-04-05' })
+        .then(results => isBillable(results, 'manytransfers@example.com'));
+    });
+
+    it('includes PILs which were transferred in and out of the establishment either side the billing period', () => {
+      return this.models.PIL.query().billable({ establishmentId: 101, start: '2017-04-06', end: '2018-04-05' })
+        .then(results => isBillable(results, 'manytransfers@example.com'));
+    });
+
+    it('does not include PILs which were transferred into the establishment after the billing period', () => {
```

Comment:
This is a dupe of the test on line 146 except for the email address - is this intentional?